src/eventstore/EventStoreSubscriber.cs

```diff
@@ -133,143 +80,156 @@ public CatchupProgress CatchUpPercentage
-                    .Result.Events.First().OriginalEventNumber;
-
-                return new CatchupProgress(_eventsProcessed, _startingPosition ?? 0, _streamName, totalEvents);
+                var lastStreamPosition = GetLastStreamPosition();
+                return new CatchupProgress(_eventsProcessed, _startingPosition ?? 0, _streamName, lastStreamPosition);
             }
         }
 
         /// <summary>
         /// Gets a value indicating whether the view model is ready or not.
         /// </summary>
-        public bool ViewModelsReady => !_catchingUp && _lastDispatchedEventNumber >= _lastNonLiveEventNumber;
+        /// <returns>Returns true if catchup is within threshold.</returns>
+        public bool IsLive
+        {
+            get
+            {
+                if (!_isSubscribed)
+                {
+                    return false;
+                }
+
+                var catchUpProgress = CatchUpPercentage;
+                var currentPosition = catchUpProgress.StartPosition + (ulong)catchUpProgress.EventsProcessed;
+                var currentPositionFromEnd = catchUpProgress.TotalEvents - currentPosition;
+
+                return currentPositionFromEnd < _liveEventThreshold;
+            }
+        }
 
         /// <summary>
         /// Creates a live eventstore subscription.
         /// </summary>
-        /// <param name="connection">Eventstore connection.</param>
+        /// <param name="eventStoreClient">Eventstore connection.</param>
         /// <param name="dispatcher">Dispatcher.</param>
         /// <param name="streamName">Stream name to push events into.</param>
         /// <param name="logger">Logger.</param>
-        /// <param name="upperQueueBound">Upper Queue Bound.</param>
-        /// <param name="heartbeatFrequency">Frequency of heartbeat.</param>
-        /// <param name="heartbeatTimeout">Timeout of heartbeat.</param>
+        /// <param name="liveEventThreshold">Proximity to end of stream before subscription considered live.</param>
         /// <returns>A new EventStoreSubscriber object.</returns>
         // ReSharper disable once UnusedMember.Global
         public static EventStoreSubscriber CreateLiveSubscription(
-            IEventStoreConnection connection,
+            EventStoreClient eventStoreClient,
             IDispatcher<ResolvedEvent> dispatcher,
             string streamName,
             ILogger logger,
-            int upperQueueBound = 2048,
-            TimeSpan? heartbeatFrequency = null,
-            TimeSpan? heartbeatTimeout = null)
-            => new EventStoreSubscriber(connection, dispatcher, streamName, logger, upperQueueBound, heartbeatFrequency, heartbeatTimeout);
+            ulong liveEventThreshold = 10)
+            => new EventStoreSubscriber(eventStoreClient, dispatcher, streamName, logger, liveEventThreshold);
 
 #pragma warning disable CS0618 // Type or member is obsolete
```

Comment:
Can we get rid of this warning disable now as I think we got rid of the obsolete tag?

Reply:
Removed still builds